### DISCUSSION TITLE
Don't zip artifact downloads

### DISCUSSION THREAD

**Discussion by brianjmurrell** [community/community#38706] [Actions] [answered]
### Select Topic Area

Product Feedback

### Body

It would be really nice if workflow artifact downloads were not always forced into .zip files.

It's super annoying to download a logfile for example and have to always run it through the zip-extract process before being able to look at it.  This really impedes one's (NPI) workflow (not GH workflow, but personal workflow).

It would also be nice to be able to download a single file from a "group file" upload where more than file were uploaded in an artifact.

**Comment by JojOatXGME** [comment]
Issue actions/upload-artifact#14 might be related. A link to this topic was recently posted there. It would be good if we could also directly open the `index.html` from an artifact (not only download it). For example, it would be nice if there were an endpoint which could be used to open or download a specific file from an artifact. For example:

    https://githubusercontent.com/<user>/<repo>/suites/<id>/artifacts/<artifact_id>/files/<relative_path_in_artifact>

**Comment by ackvf** [comment]
On the other hand, I have a matrix workflow for each platform, the result is a zip file. I then want to Download the artifacts for a release and when I download the zips, they get automatically extracted and deleted, causing other issues in the pipeline.

How do I prevet `actions/download-artifact@v4` from extracting the archives?

**Comment by nebuk89** [comment, marked as answer]
https://github.blog/changelog/2026-02-26-github-actions-now-supports-uploading-and-downloading-non-zipped-artifacts/
Shipped <3

**Comment by myhau** [comment]
@nebuk89 

Thank you so much for shipping this! 🙏

Are there any plans for supporting other mime types like video or gif?
